app/models/concerns/user_reports.rb

```diff
@@ -0,0 +1,20 @@
+module UserReports
+  extend ActiveSupport::Concern
+
+  def reports
+    return Types::Report.values if user_manager? && FeatureFlags.allow_user_managers_service_access.enabled?
+    return [] if user_manager?
+
+    Types::USER_ROLE_REPORTS.fetch(role)
+  end
```

Comment:
At the moment if `.fetch` fails because of some dodgy role the exception will be an obtuse error about invalid hash or something. Should there be an explicit check beforehand e.g. `raise LaaException("Unsupported role: '#{role}'") unless Types::USER_ROLE_REPORTS.key?(role)` - i suppose thats an issue for all role access code!

Reply:
I've updated the hash so that it uses the types for the keys, which should hopefully mean that this cannot happen. (I was using constants before, so unlikely, but I think better via the type.)

```
  USER_ROLE_REPORTS = {
    UserRole[CASEWORKER_ROLE] => [Report['workload_report'], Report['processed_report']],
    UserRole[SUPERVISOR_ROLE] => Report.values
  }.freeze
```